common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -145,6 +149,18 @@ public Collation(
     }
   }
 
+  /**
+   * Auxiliary methods for collation aware string operations.
+   */
+
+  public static StringSearch getStringSearch(final UTF8String left, final UTF8String right,
+                                          final int collationId) {
```

Comment:
```suggestion
  public static StringSearch getStringSearch(
      final UTF8String left,
      final UTF8String right,
      final int collationId) {
```

Reply:
are we going to use it for startsWith and others?

Reply:
definitely, that's in ([SPARK-47295](https://issues.apache.org/jira/browse/SPARK-47295)) and will serve as an onboarding task for a new eng (of course, I will help them out with this)